packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1130 @@
+import * as React from 'react';
+import { IComboBoxProps, IComboBoxOption } from './ComboBox.Props';
+import { DropdownMenuItemType } from '../../Dropdown';
+import { DirectionalHint } from '../../common/DirectionalHint';
+import { Callout } from '../../Callout';
+import { Label } from '../../Label';
+import {
+  CommandButton,
+  IconButton
+} from '../../Button';
+import { DynamicAutoFill } from '../pickers/AutoFill/DynamicAutoFill/DynamicAutoFill';
+import { BaseAutoFill } from '../pickers/AutoFill/BaseAutoFill';
+import {
+  BaseComponent,
+  KeyCodes,
+  autobind,
+  css,
+  findIndex,
+  getId
+} from '../../Utilities';
+import * as stylesImport from './ComboBox.scss';
+const styles: any = stylesImport;
+
+export interface IComboBoxState {
+
+  // The open state
+  isOpen?: boolean;
+
+  // The currently selected index (-1 if no index is selected)
+  selectedIndex?: number;
+
+  // The focused state of the comboBox
+  focused?: boolean;
+
+  // This value is used for the autocomplete hint value
+  suggestedDisplayValue?: string;
```

Comment:
actually, this should be a string with the value. There's no guarantee that the suggested display value maps to an index